.github/workflows/lint.yml

```diff
@@ -125,7 +125,64 @@ jobs:
         working-directory: ./
         run: npm run ci:all
 
-      # TODO: Enable in lint fix PR
+      # TODO: Enable post eslint prettier configuration
       # - name: Linting
       #   working-directory: ./
       #   run: npm run prettier:all
+
+  # 4. Type check
+  typecheck:
+    name: TypeScript ⌨️
+    environment:
+      name: ${{ needs.setup.outputs.environment }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Timezone
+        uses: szenius/set-timezone@v2.0
+        with:
+          timezoneLinux: ${{ needs.setup.outputs.timezone }}
+
+      - name: Repository
+        uses: actions/checkout@v4
+
+      - name: Node
+        uses: actions/setup-node@v4
+        with:
+          node-version: ${{ vars.NODE_VERSION }}
+
+      - name: Dependencies
+        working-directory: ./
+        run: npm run ci:all
+
+      # TODO: Enable post TSC error resolution
```

Comment:
Do we need a ticket for this?

Reply:
Yes we do, would you like me to add ticket numbers to all TODO?

Reply:
Have added ticket numbers to `TODO`s. Thanks